Assets/Plugins/Source/Editor/ConfigEditors/ConfigEditor.cs

```diff
@@ -22,29 +22,146 @@
 
 namespace PlayEveryWare.EpicOnlineServices.Editor
 {
-    using System.Threading.Tasks;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using UnityEditor;
+    using UnityEngine;
+    using Utility;
+    using Task = System.Threading.Tasks.Task;
 
     /// <summary>
     /// Contains implementations of IConfigEditor that are common to all implementing classes.
     /// </summary>
     /// <typeparam name="T">Intended to be a type accepted by the templated class EOSConfigFile.</typeparam>
-    public abstract class ConfigEditor<T> : IConfigEditor where T : EpicOnlineServices.Config
+    public class ConfigEditor<T> : IConfigEditor where T : EpicOnlineServices.Config
     {
         private readonly string _labelText;
         protected T config;
 
+        public ConfigEditor()
+        {
+            Type configType = typeof(T);
+
+            ConfigGroupAttribute attribute = configType.GetCustomAttribute<ConfigGroupAttribute>();
+
+            if (null != attribute)
+            {
+                _labelText = attribute.Label;
+            }
+        }
+
         protected ConfigEditor(string labelText)
         {
             _labelText = labelText;
         }
 
+        /// <summary>
+        /// Use reflection to retrieve a collection of fields that have been
+        /// assigned custom ConfigFieldAttribute attributes, grouping by group,
+        /// and sorting by group.
+        /// </summary>
+        /// <returns>A collection of config fields.</returns>
+        private static IOrderedEnumerable<IGrouping<int, (FieldInfo FieldInfo, ConfigFieldAttribute FieldDetails)>> GetFieldsByGroup()
+        {
+            var returnValue = typeof(T).GetFields(BindingFlags.Public | BindingFlags.Instance)
+                .Where(field => field.GetCustomAttribute<ConfigFieldAttribute>() != null)
+                .Select(info => (info, info.GetCustomAttribute<ConfigFieldAttribute>()))
```

Comment:
In the parameterless constructor, it attempts to get this `ConfigGroupAttribute`, and then there's a null check against it. Here, it selects it and uses it without null checking. Is that okay? Is there ever a circumstance where this can be null, and that breaks the rest of this LINQ query?

Reply:
Nope - the reason is that the `Where` clause excludes any results for which the field is `null`, meaning only those that are not null are subsequently processed by the `Select` clause.

Reply:
Ah, I was thrown off a bit by the LINQ naming. The `.Where` uses `field` where the `.Select` uses `info`, so I read that as different comparisons. Seems good!